### PR TITLE
Set sameSite attributes on Konductor cookies

### PR DESCRIPTION
**Pull request by jonsnyder**
## Description

Konductor is adding attributes `sameSite` and `secure` to cookies returned in the state:store handle. This change honors those attributes. Anytime sameSite is set to none, the secure flag is added because that is required by browsers for the cookie to work.

I attempted to create a failing functional test to show how this fixes an issue with cross-domain iframe, but the testCafe hammerhead proxy doesn't accurately reflect browser behavior for cross-domain iframe. I've committed this change [here](https://github.com/adobe/alloy/commit/4f4c822b27eabbc467ce6e5c421f78bd2d6b46f5), but not added it to this PR because it doesn't actually fail before this fix. Also it would require changes to our alloyio.com site to include an additional domain name and cert. See https://github.com/adobe/alloy/wiki/Running-the-sandbox-locally-over-HTTPS-on-Mac-OS for instructions on how to test cross-domain requests manually.

## Related Issue

https://jira.corp.adobe.com/browse/PDCL-7396

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

File: sandbox/public/iframed.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      iframe {
+        height: 100%;
+      }
+      iframe {
+        width: 100%;
+      }
+    </style>
+  </head>
+  <body>
+    <iframe id="iframe" src="https://alloyio.com:3000" />
```
Comment (jfkhoury) [inline sandbox/public/iframed.html:18]: Will this work once we host this file on `alloyio.com`? Are you trying to create a cross domain situation?

File: src/core/createCookieTransfer.js [hunk 2]
```diff
@@ -67,11 +67,18 @@ export default ({ cookieJar, orgId, apexDomain }) => {
...
+        const sameSite = stateItem.sameSite && stateItem.sameSite.toLowerCase();
 
         if (stateItem.maxAge !== undefined) {
           // cookieJar expects "expires" in days
           options.expires = convertTimes(SECOND, DAY, stateItem.maxAge);
         }
+        if (sameSite !== undefined) {
+          options.sameSite = sameSite;
+        }
+        if (sameSite === "none" || stateItem.secure === true) {
```
Comment (jfkhoury) [inline src/core/createCookieTransfer.js:79]: Should we add a comment that we don't really expect the Edge to set the `secure` attribute, so that's why we're setting it? Just for future eyes :)

**Review by jfkhoury — APPROVED** [review]
Sweet!!

Should we add a functional test now that Konductor has released this feature?

**Review by ninaceban — APPROVED** [review]
👍

**Comment by jonsnyder** [comment]
There's no way to add a functional test because you cannot inspect what attributes were set for a cookie.

File: sandbox/public/iframed.html (hunk 1 above)
Comment (jonsnyder) [reply sandbox/public/iframed.html:18]: This is meant to be used locally. Yes, I'm trying to create a cross domain iframe. https://github.com/adobe/alloy/wiki/Running-the-sandbox-locally-over-HTTPS-on-Mac-OS

Comment (ninaceban) [reply sandbox/public/iframed.html]: Maybe we could use a dummy GitHub page in the iFrame, that would work for both local and hosted on alloyio.com?

Comment (jonsnyder) [reply sandbox/public/iframed.html]: I removed the port number. I'm still able to test by hosting alloyio.com locally on port 443. I added instructions to the wiki: https://github.com/adobe/alloy/wiki/Running-the-sandbox-locally-over-HTTPS-on-Mac-OS

File: src/components/Audiences/injectProcessDestinations.js [hunk 3]
```diff
@@ -10,48 +10,59 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
-    dest => dest.type === "cookie"
-  );
+export default ({
+  fireReferrerHideableImage,
+  logger,
+  cookieJar,
+  isPageSsl
+}) => {
+  const extraCookieOptions = isPageSsl
+    ? { sameSite: "none", secure: true }
```
Comment (ninaceban) [inline src/components/Audiences/injectProcessDestinations.js:28]: Aren't these options returned by the Konductor if it's https?

Comment (jonsnyder) [reply src/components/Audiences/injectProcessDestinations.js:28]: They are for "state:store" handles, but not "activation:push" handles.

**Comment by jonsnyder** [comment]
We found issues with identity cookies when the user visits a secure page and then a non-secure page. We need to add a functional test with this use case, and we need to modify the behavior of Experience Edge before releasing this.

**Comment by jonsnyder** [comment]
> > We found issues with identity cookies when the user visits a secure page and then a non-secure page. We need to add a functional test with this use case, and we need to modify the behavior of Experience Edge before releasing this.
> 
> @jonsnyder What's the latest on these issues?

We decided to go ahead with this change. We should encourage customers to host all their pages securely.
